Source/EasyNetQ/PersistentConnection.cs

```diff
@@ -1,195 +1,111 @@
 
-        private void StartTryToConnect()
+        public bool IsConnected => connection.IsOpen;
+
+        public void Dispose()
         {
-            Timers.RunOnce(s => TryToConnect(), connectionFactory.Configuration.ConnectIntervalAttempt);
+            connection.Dispose();
         }
 
-        private void TryToConnect()
+        private IConnection InitializeConnection()
         {
-            logger.Debug("Trying to connect");
-            if (disposed) return;
-
-            connectionFactory.Reset();
-            do
-            {
-                try
-                {
-                    connection = connectionFactory.CreateConnection(); // A possible dispose race condition exists, whereby the Dispose() method may run while this loop is waiting on connectionFactory.CreateConnection() returning a connection.  In that case, a connection could be created and assigned to the connection variable, without it ever being later disposed, leading to app hang on shutdown.  The following if clause guards against this condition and ensures such connections are always disposed.
-                    if (disposed)
-                    {
-                        connection.Dispose();
-                        break;
-                    }
-
-                    try
-                    {
-                        OnConnected();
-                    }
-                    catch
-                    {
-                        connection.Dispose();
-                        throw;
-                    }
-                    
-                    connection.ConnectionShutdown += OnConnectionShutdown;
-                    connection.ConnectionBlocked += OnConnectionBlocked;
-                    connection.ConnectionUnblocked += OnConnectionUnblocked;
-
-                    logger.InfoFormat(
-                        "Connected to broker {broker}, port {port}, vhost {vhost}",
-                        connectionFactory.CurrentHost.Host,
-                        connectionFactory.CurrentHost.Port,
-                        connectionFactory.Configuration.VirtualHost
-                    );
-
-                    connectionFactory.Success();
-                }
-                catch (Exception ex) when (ex is SocketException || ex is BrokerUnreachableException || ex is TimeoutException)
-                {
-                    LogException(ex);
-                }
-            } while (!disposed && connectionFactory.Next());
-
-            if (!connectionFactory.Succeeded && !disposed)
+            var endpoints = connectionConfiguration.Hosts.Select(x =>
             {
-                logger.ErrorFormat("Failed to connect to any Broker. Retrying in {connectInterval}", connectionFactory.Configuration.ConnectIntervalAttempt);
-                StartTryToConnect();
-            }
+                var endpoint = new AmqpTcpEndpoint(x.Host, x.Port);
+                if (x.Ssl.Enabled)
```

Comment:
I prefer if statement in this way:

```csharp
if (x.Ssl.Enabled)
{
    endpoint.Ssl = x.Ssl;
}
else if (connectionConfiguration.Ssl.Enabled)
{
    endpoint.Ssl = connectionConfiguration.Ssl;
}
```

like the Allman style rule enforced on the [corefx development](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md) (point 1):

> We use Allman style braces, where each brace begins on a new line. A single line statement block can go without braces but the block must be properly indented on its own line and must not be nested in other statement blocks that use braces (See issue 381 for examples). One exception is that a using statement is permitted to be nested within another using statement by starting on the following line at the same indentation level, even if the nested using contains a controlled block.

Reply:
I prefer the minimalist/functional style. Less braces = less clutter :)

Reply:
It could open to strange and sneaky bugs, see this discussion: https://github.com/dotnet/corefx/issues/381

BTW is not so important in this review 😄 but I wanted to point out that is not just syntactic sugar problem.

Reply:
I hope we have to deal about the code style. 

BTW I used default rider/r# code style, except "_" name prefix of fields.